sm/tedge_agent/src/agent.rs

```diff
@@ -345,29 +367,130 @@ impl SmAgent {
         Ok(())
     }
 
-    async fn fail_pending_operation(&self, mqtt: &Client) -> Result<(), AgentError> {
+    async fn handle_restart_operation(
+        &self,
+        mqtt: &Client,
+        message: &Message,
+    ) -> Result<(), AgentError> {
+        let request = match RestartOperationRequest::from_slice(message.payload_trimmed()) {
+            Ok(request) => {
+                let () = self
+                    .persistance_store
+                    .store(&State {
+                        operation_id: Some(request.id.clone()),
+                        operation: Some(StateStatus::Restart(RestartOperationStatus::Restarting)),
+                    })
+                    .await?;
+                request
+            }
+
+            Err(error) => {
+                error!("Parsing error: {}", error);
+                let _ = mqtt
+                    .publish(Message::new(
+                        &self.config.errors_topic,
+                        format!("{}", error),
+                    ))
+                    .await?;
+
+                return Err(SoftwareError::ParseError {
+                    reason: "Parsing failed".into(),
+                }
+                .into());
+            }
+        };
+
+        self.persistance_store
+            .update(&StateStatus::Restart(RestartOperationStatus::Restarting))
+            .await?;
+
+        let () = restart_operation::create_slash_run_file()?;
```

Comment:
An error here will not be neither reported to the caller nor stored in the persistent state.

* This is the case for most of the errors raised in the `handle_restart_operation`.
* A `?` is akin to throwing an exception. One need a place to catch these errors.